config.py.example

```diff
@@ -0,0 +1,122 @@
+########################################
+# APP-SPECIFIC CONFIGURATION VARIABLES #
+########################################
+# The following variables should or must be defined in your
+# own application. Create a config.py file, import that config_common
+# file as:
+#
+#   from config_hub import *
+#
+# then define the following variables to fit your needs. You can also override any
+# any other variables in this file as required. Variables defined as ValueError() exceptions
+# *must* be defined
+#
+
+import logging
+
+from biothings.utils.loggers import setup_default_log
+from biothings.utils.configuration import ConfigurationError, ConfigurationDefault, ConfigurationValue
+
+
+#* 7. Hub Internals *#
+DATA_SRC_SERVER = ConfigurationError("Define hostname for source database")
+DATA_SRC_PORT = ConfigurationError("Define port for source database")
+DATA_SRC_DATABASE = ConfigurationError("Define name for source database")
+DATA_SRC_SERVER_USERNAME = ConfigurationError("Define username for source database connection (or None if not needed)")
+DATA_SRC_SERVER_PASSWORD = ConfigurationError("Define password for source database connection (or None if not needed)")
+
+# Target (merged collection) database connection
+DATA_TARGET_SERVER = ConfigurationError("Define hostname for target database (merged collections)")
+DATA_TARGET_PORT = ConfigurationError("Define port for target database (merged collections)")
+DATA_TARGET_DATABASE = ConfigurationError("Define name for target database (merged collections)")
+DATA_TARGET_SERVER_USERNAME = ConfigurationError("Define username for target database connection (or None if not needed)")
+DATA_TARGET_SERVER_PASSWORD = ConfigurationError("Define password for target database connection (or None if not needed)")
+
+HUB_DB_BACKEND = ConfigurationError("Define Hub DB connection")
+# Internal backend. Default to mongodb
+# For now, other options are: mongodb, sqlite3, elasticsearch
+#HUB_DB_BACKEND = {
+#        "module" : "biothings.utils.sqlite3",
+#        "sqlite_db_foder" : "./db",
+#        }
+#HUB_DB_BACKEND = {
+#        "module" : "biothings.utils.mongo",
+#        "uri" : "mongodb://localhost:27017",
+#        #"uri" : "mongodb://user:passwd@localhost:27017", # mongodb std URI
+#        }
+#HUB_DB_BACKEND = {
+#        "module" : "biothings.utils.es",
+#        "host" : "localhost:9200",
+#        }
+
+# cache file format ("": ascii/text uncompressed, or "gz|zip|xz"
+CACHE_FORMAT = "xz"
```

Comment:
settings like `CACHE_FORMAT`, `BIOTHINGS_ROLE` and others below are not found in the `default_config.py`. They should be defined in `default_config.py` and removed from this `config.py.example`, right? Most hubs won't change them. There should not be any sdk-level settings here not be defined in default.

At the end of this file, you can add a commented section for any possible user-hub specific settings if needed.

Reply:
It's updated